src/transformers/generation/flax_utils.py

```diff
@@ -793,6 +837,10 @@ def gather_fn(tensor):
         eos_token_id = eos_token_id if eos_token_id is not None else self.generation_config.eos_token_id
         length_penalty = length_penalty if length_penalty is not None else self.generation_config.length_penalty
         early_stopping = early_stopping if early_stopping is not None else self.generation_config.early_stopping
+        output_scores = output_scores if output_scores is not None else self.generation_config.output_scores
+        num_return_sequences = (
+            num_beam_hyps_to_keep if num_beam_hyps_to_keep is not None else self.generation_config.num_return_sequences
```

Comment:
E.g. here, if `num_beam_hyps_to_keep` defaults to 1, then we'll always default to 1 beam to keep (whereas it should default to None, and we should use `generation_config.num_return_sequences`)